src/contours.rs

```diff
@@ -0,0 +1,193 @@
+//! Functions for detecting contours of polygons in an image and approximating
+//! polygon from set of points.
+
+use crate::definitions::Point;
+use image::GrayImage;
+use num::{cast, Num, NumCast};
+use std::collections::VecDeque;
+
+/// Contour struct containing its points, is_outer flag to determine whether the
+/// contour is an outer border or hole border, and the parent option (the index
+/// of the parent contour in the contours vec)
+#[derive(Debug)]
+pub struct Contour<T: Num + NumCast + Copy + PartialEq + Eq> {
+    /// All the points on the contour
```

Comment:
Nit: doc text lines should end with a '.' as otherwise it looks odd in the rendered docs.